lib/util/auth.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * @typedef {require('../metadataTypes/MetadataType.js')} MetadataType
+ */
+
+/**
+ * @typedef {Object} BuObject
+ * @property {String} credential instance identifier
+ * @property {Integer} mid ID of Business Unit to authenticate with
+ * @property {Integer} eid Parent ID of Business Unit to authenticate with
+ * @property {String} businessUnit name of Business Unit to authenticate with
```

Comment:
simple types (string, integer, boolean) should be lower-case
complex types (Object, Array) should be upper-case

mixed stuff like this: "string[]"